src/sqlfluff/dialects/dialect_oracle.py

```diff
@@ -0,0 +1,90 @@
+"""The Oracle dialect.
+
+This inherits from the ansi dialect.
+"""
+from sqlfluff.core.dialects import load_raw_dialect
+from sqlfluff.core.parser.segments import NewlineSegment
+from sqlfluff.core.parser.segments.raw import KeywordSegment
+from sqlfluff.core.parser import (
+    Anything,
+    BaseSegment,
+    CommentSegment,
+    Ref,
+    Sequence,
+    StartsWith,
+    StringParser,
+    OptionallyBracketed,
+    OneOf,
+)
+
+ansi_dialect = load_raw_dialect("ansi")
+oracle_dialect = ansi_dialect.copy_as("oracle")
+
+oracle_dialect.sets("unreserved_keywords").difference_update(["COMMENT"])
+oracle_dialect.sets("reserved_keywords").update(["ON"])
+
+oracle_dialect.add(
+        CommentKeywordSegment=StringParser(
+            "COMMENT", KeywordSegment, name="comment"
+        )
+)
```

Comment:
This wouldn't be the way to do this. You would only "add" a segment like this if you want to add a recognisable "block".

For example:
```sql
-- This is a comment
```

This might be handled a as one "block", starting with `--` and then including any new line characters until then end of the line.

Here the `COMMENT` is just a keyword, as is covered above, so you don't need this code at all, so remove it.

```suggestion
```